internal/index/indexoperations/index.go

```diff
@@ -52,3 +54,18 @@ func ListIndexes(path string) ([]Index, error) {
 	}
 	return indexes, nil
 }
+
+// AddIndex initializes a new index to install plugins from.
+func AddIndex(path, name, url string) error {
+	if strings.ContainsAny(name, "./") {
+		return errors.New("index name cannot contain path characters")
+	}
+
+	dir := filepath.Join(path, name)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		return gitutil.EnsureCloned(url, dir)
+	} else if err != nil {
+		return errors.Wrapf(err, "failed to describe %s", dir)
```

Comment:
- `describe` &rarr; `read directory`

- %s is not necessary, as file errors always contain the path anyway.

Reply:
I assume that the go error already has a sufficient human-readable description of the error. So maybe wrapping is just unnecessary here.